src/app/modules/shared_models/message_model.nim

```diff
@@ -603,7 +603,10 @@ QtObject:
 
   proc pinUnpinMessage*(self: Model, messageId: string, pin: bool, pinnedBy: string) =
     let ind = self.findIndexForMessageId(messageId)
-    if(ind == -1):
+    if ind == -1:
+      return
+
+    if self.items[ind].pinned == pin:
```

Comment:
I don't understand this chunk :)

Reply:
Yeah, the naming is not great. I'll rename `pin` to `isPinned`.
So this chunk is just a normal check we usually on the models that we didn't have before.
It doesn't affect the fix at all, I just discovered that we call `dataChanged` here even if the message `pinned` status didn't change